### ISSUE TITLE
Steam Deck not compatible

### ISSUE THREAD

**Issue by luiscesjr** [TokisanGames/Terrain3D#220]
Just like the issue I have in #197 , when running my game on the steam deck any level that contain Terrain3D won't work.

The game does not crash, it just does not open the scene.

I have Jolt in the project and it runs fine, so I must assume it's not a GDextension issue like the 197 problem on Android.

I'm assuming this is a steam deck problem only, since there are no Linux compatibility errors

Additional info:

- SteamOS 3.5
- Godot 4.2 dev1
- Godot export for Linux, I'm not using Proton to layer windows binaries. I'll try that later tough.
- Latest Terrain3D release

Unfortunately it's hard to debug in the deck since there is no official way to run the game on it, apart from using Valve's devkit software to send over the game.

**Comment by TokisanGames** [comment]
Ok, thanks for the report. You're going to have to do the legwork to test and troubleshoot. That probably means building the plugin so you can tweak code. I can provide upstream support. The latest terrain3d release is only tested on 4.1.1. I have not done any testing on 4.2. Can you get any console to look at logs?

If it doesn't crash and doesn't open the scene, what does it do? Assuming you are running it from a launcher, how does it not open the scene but that's not a crash on startup?

Have you tested the demo?

Are you sure the terrain3d libraries were copied to the device along with the executable?

**Comment by luiscesjr** [comment]
> Ok, thanks for the report. You're going to have to do the legwork to test and troubleshoot. That probably means building the plugin so you can tweak code. I can provide upstream support. The latest terrain3d release is only tested on 4.1.1. I have not done any testing on 4.2. Can you get any console to look at logs?

No problem, I'll get to test it further this week. It works very well for me on 4.2 dev1 - dev5, I only use dev1 because of a problem on another plugin, Nakama, which there were some changes to http requests on dev2 that broke it.

> If it doesn't crash and doesn't open the scene, what does it do? Assuming you are running it from a launcher, how does it not open the scene but that's not a crash on startup?

My game opens on a login screen first, then main menu. So only after I select a level with Terrain3D it simply does not load, stays on the menu without crashing.

> Have you tested the demo?

Not yet, will do soon.

> Are you sure the terrain3d libraries were copied to the device along with the executable?

Yes, made sure they got all sent, and are there.

**Comment by jacobcoughenour** [comment]
I built Terrain3D at 3c460256f4b0596785071e72cb3d90a45d55b5a9 on my steamdeck with godot-cpp at [17137b2](https://github.com/godotengine/godot-cpp/commit/17137b2e2ecdeb5638a885d28498164d35a2130d) and opened the demo project in Godot 4.2.stable and installed godot-jolt v0.11.0-stable into the demo project.
I'm also on Steam OS v3.5.7 and I had to install glibc and linux-api-headers to get it to build in addition to the usual godot dependencies.

The demo runs just fine running it from the editor and exporting it so perhaps the fix for #197 also fixed this issue or something changed in godot, godot-jolt, etc that fixed it.

Let me know if there is anything else I should try now that I have a proper debugging setup on my steamdeck.

Also if anyone is curious, I'm easily getting 200+ fps on the exported demo project on my first generation steamdeck.

![Image](https://github.com/TokisanGames/Terrain3D/assets/5546400/47705cf6-3344-47b6-8ffc-188d1f5d8e50)
